src/evm/protocol/uniswap_v4/hooks/generic_vm_hook_handler.rs

```diff
@@ -557,4 +627,97 @@ mod tests {
         // This hook does not return any delta, so we expect it to be zero.
         assert_eq!(res, I128::ZERO);
     }
+
+    #[test]
+    fn test_get_amount_ranges() {
+        let block = BlockHeader {
+            number: 15797251,
+            hash: Bytes::from_str(
+                "0x7032b93c5b0d419f2001f7c77c19ade6da92d2df147712eac1a27c7ffedfe410",
+            )
+            .unwrap(),
+            timestamp: 1748397011,
+            ..Default::default()
+        };
+        let db = SimulationDB::new(get_client(None), get_runtime(), Some(block.clone()));
+        let engine = create_engine(db, true).expect("Failed to create simulation engine");
+
+        let hook_address = Address::from_str("0x0010d0d5db05933fa0d9f7038d365e1541a41888")
+            .expect("Invalid hook address");
+
+        let pool_manager = Address::from_str("0x000000000004444c5dc75cB358380D2e3dE08A90")
+            .expect("Invalid pool manager address");
+
+        // The limits contract address from the provided entrypoint
+        let limits_contract_address =
+            Address::from_str("0x69058613588536167ba0aa94f0cc1fe420ef28a8")
+                .expect("Invalid limits contract address");
+
+        // Deploy mock getLimits contract to avoid intricacies of existing contracts
+        //
+        // Bytecode retrieved by running `forge inspect Limits deployedBytecode` on the following
+        // contract (must be converted to a Solidity file):
+        //
+        // // SPDX-License-Identifier: UNLICENSED
+        // pragma solidity ^0.8.26;
+        //
+        // contract Limits {
+        //     function getLimits(address a, address b) external pure returns (uint256, uint256) {
+        //         require(a != address(0) && b != address(0));
+        //         return (1000, 2000);
+        //     }
+        // }
```

Comment:
aaaahhhhhh sneaky.. I think this is ok to have as a test. But can we add a test for the real get_limits from Euler please?